### DISCUSSION TITLE
Confused as to why Partial<MyContext> is being used everywhere.

### DISCUSSION THREAD

**Discussion by jcheroske** [statelyai/xstate#1688] [Q&A] [answered]
I'm unclear why xstate invokes actions, guards, delays, etc. with the type of the context argument set to `Partial<MyContext>` instead of just `MyContext`. I'm having to deal with the possibility of undefined everywhere, even though there is no way it can actually occur. I'm a typescript noob, so any help is much appreciated. I'm probably missing the obvious.

**Comment by Andarist** [comment]
Could you demo your problem with a simple codesandbox? I don't see `Partial` being used in a lot of places and those that I can see shouldn't result in a problem that you are describing.

**Comment by jcheroske** [comment, marked as answer]
Sorry to have wasted your time. I found the actual problem lurking in my code. Thank you for all your work on this great library!
